### PR TITLE
[RFC] strip debugging symbols from musl and glibc

### PR DESCRIPTION
**Pull request by CameronNemo**
[ci skip]

### CODE REVIEW

**Comment by CameronNemo** [comment]
xbps-src will refuse a non-debug package depending on debug package.

**Comment by newbluemoon** [comment]
Just for reference, there was a problem once with missing libc debugging symbols in the old repository and a comment on why they’re there in the first place:

https://github.com/voidlinux/void-packages/issues/7734#issuecomment-346613763

**Comment by CameronNemo** [comment]
Memory requirements were not what I had in mind. I was comparing disk usage of the debian buster-slim docker image and void's image. Stripping debug symbols saves more than 20 MB (uncompressed) on glibc, which is quite significant considering the slim Debian images are ~70 MB uncompressed.

As for the valgrind/gdb use case: I think these users are more than capable of installing debug packages. Having these tools work out of the box is not strictly necessary for a functioning system. It is only slightly more convenient.

I would appreciate your input here @chneukirchen.

**Comment by maxice8** [comment]
@void-linux/pkg-committers

File: srcpkgs/glibc/template [hunk 1]
```diff
@@ -10,38 +10,6 @@ homepage="http://www.gnu.org/software/libc"
...
-nostrip_files="
-	XBS5_ILP32_OFFBIG
-	XBS5_ILP32_OFF32
-	POSIX_V6_ILP32_OFF32
-	POSIX_V6_ILP32_OFFBIG
-	POSIX_V7_ILP32_OFF32
-	POSIX_V7_ILP32_OFFBIG
-	POSIX_V6_LP64_OFF64
-	POSIX_V7_LP64_OFF64
-	XBS5_LP64_OFF64
```
Comment (Gottox) [inline srcpkgs/glibc/template]: are those files?

**Comment by chneukirchen** [comment]
I don't like the change but if you want to do it, go ahead.

**Comment by Gottox** [comment]
@chneukirchen what are the downsides of this change?

File: srcpkgs/glibc/template (hunk 1 above)
Comment (CameronNemo) [reply srcpkgs/glibc/template]: they are shared objects in `/usr/lib/gconv` and `/usr/lib/getconf` not sure what they are for exactly.

Comment (Gottox) [reply srcpkgs/glibc/template]: But xbps-src doesn't scan for symbols, only for files.

Comment (CameronNemo) [reply srcpkgs/glibc/template]: Okay on second look there are executables in /usr/lib/getconf that match these filenames.

Comment (Gottox) [reply srcpkgs/glibc/template]: ah!

**Comment by CameronNemo** [comment]
@leahneukirchen are you concerned that this will break jq's test suite?

**Comment by leahneukirchen** [comment]
No, I don't care about jq's test suite.

It will make debugging in the wild more annoying, that is all.

**Comment by thypon** [comment]
@leahneukirchen maybe we can have it stripped and included by default in the base-system and not in the base-minimal.

**Comment by CameronNemo** [comment]
Pretty sure that a package in the main repo can't depend on a package in the debug repo. So we would need two package versions, one stripped and one not. That could make transitions more complicated and I'm not sure it is worth it.

**Comment by maxice8** [comment]
ping

**Comment by Gottox** [comment]
@CameronNemo why do we need an unstripped version?

**Comment by CameronNemo** [comment]
If we wanted base-system to have debugging symbols and base-minimal to not. I was responding to @thypon 's suggestion but that was not clear.

**Comment by the-maldridge** [comment]
I think with the launch of deboginfod this should be more palatable.  If you're doing a debug trace in the wild you're likely already going to have network access to fetch the debug packages for whatever you're actually debugging.   With debuginfod this should now be fully transparent as symbols are fetched on the fly.

**Pull request closed without merging**